File: backend/src/routes/v1/upload.routes.js
import { Router } from 'express';
import UploadController from '../../controllers/upload.controller.js';
import { authenticate } from '../../middleware/auth.middleware.js';
import upload from '../../middleware/multer.middleware.js';

const router = Router();

router.use(authenticate);

router.post('/image', upload.single('file'), UploadController.uploadImage);
router.post('/document', upload.single('file'), UploadController.uploadDocument);

export default router;

File: ClearNest-main/backend/app/routes/upload_routes.py
from flask import Blueprint
from app.controllers.upload_controller import UploadController
from flask_jwt_extended import jwt_required

upload_bp = Blueprint('upload', __name__)

@upload_bp.route('/image', methods=['POST'])
@jwt_required()
def upload_image():
    return UploadController.upload_image()

@upload_bp.route('/document', methods=['POST'])
@jwt_required()
def upload_document():
    return UploadController.upload_document()
